components/builder-web/app/actions/packages.ts

```diff
@@ -68,12 +78,38 @@ export function fetchPackage(pkg) {
         dispatch(clearPackages());
         depotApi.get(pkg.ident).then(response => {
             dispatch(setCurrentPackage(response["results"]));
+            dispatch(fetchCurrentPackageChannels(pkg));
         }).catch(error => {
             dispatch(setCurrentPackage(undefined, error));
         });
     };
 }
 
+export function fetchCurrentPackageChannels(pkg) {
+    return dispatch => {
+        dispatch(clearPackageChannels());
+        depotApi.getPackageChannels(pkg.ident)
+            .then(channels => dispatch(setCurrentPackageChannels(channels)))
+            .catch(error => dispatch(setCurrentPackageChannels(undefined, error)));
+    };
+}
+
+export function fetchVisiblePackageChannels(pkgs) {
+    return dispatch => {
+        pkgs.forEach((pkg) => {
+
+            // Only fetch for fully qualified packages
+            if (pkg.origin && pkg.name && pkg.version && pkg.release) {
+                depotApi.getPackageChannels(pkg)
+                    .then(channels => {
+                        dispatch(setVisiblePackageChannels(pkg, channels));
+                    })
+                .catch(error => console.error(error));
+            }
+        });
+    };
+}
```

Comment:
Here is a quick POC that I did just now to show what I was talking about on your PR, as an alternative to making N+1 HTTP requests to fetch channels for packages.  It's pretty rough, and only implemented for listing packages, but I think it at least shows what's possible.  https://github.com/habitat-sh/habitat/compare/jb/lulz?expand=1